src/core/job.c

```diff
@@ -598,40 +598,56 @@ static const char* job_done_message_format(Unit *u, JobType t, JobResult result)
         static const char* const generic_finished_verify_active_job[_JOB_RESULT_MAX] = {
                 [JOB_SKIPPED]     = "%s is inactive.",
         };
-        const char *format;
 
         assert(u);
         assert(t >= 0);
         assert(t < _JOB_TYPE_MAX);
 
         /* Show condition check message if the job did not actually do anything due to failed condition. */
-        if (t == JOB_START && result == JOB_DONE && !u->condition_result)
-                return "Condition check resulted in %s being skipped.";
+        if (t == JOB_START && result == JOB_DONE && !u->condition_result) {
+                Condition *c;
+
+                LIST_FOREACH(conditions, c, u->conditions)
+                        if (c->result != CONDITION_SUCCEEDED)
+                                break;
+
+                assert(c);
+
+                return strjoin(
+                        "%s was skipped because of a failed condition check: ",
+                        condition_type_to_string(c->type),
+                        "=",
+                        c->trigger ? "|" : "",
+                        c->negate ? "!" : "",
+                        c->parameter);
+        }
 
         if (IN_SET(t, JOB_START, JOB_STOP, JOB_RESTART)) {
                 const UnitStatusMessageFormats *formats = &UNIT_VTABLE(u)->status_message_formats;
+                const char *format;
+
                 if (formats->finished_job) {
                         format = formats->finished_job(u, t, result);
                         if (format)
-                                return format;
+                                return strdup(format);
                 }
 
                 format = (t == JOB_START ? formats->finished_start_job : formats->finished_stop_job)[result];
                 if (format)
-                        return format;
+                        return strdup(format);
         }
 
         /* Return generic strings */
         switch (t) {
         case JOB_START:
-                return generic_finished_start_job[result];
+                return strdup(generic_finished_start_job[result]);
```

Comment:
no OOM checks?

Reply:
Well `strdup` returns `NULL` if it fails to allocate AFAIK so we can just return the result directly without having to do an OOM check.

Reply:
but are the callers expecting that and raising -ENOMEM?

Reply:
There's only one caller in this case which is `job_emit_done_message` which returns `void` and already had a `NULL` check to return early if `job_emit_done_message` returned `NULL`.